packages/client/tests/functional/issues/14001-mongo-order-by-conflict/tests.ts

```diff
@@ -0,0 +1,69 @@
+import testMatrix from './_matrix'
+// @ts-ignore
+import type { PrismaClient } from './node_modules/@prisma/client'
+
+declare let prisma: PrismaClient
+
+/**
+ * Regression test for #14001
+ * Invalid character in mongo query
+ */
+testMatrix.setupTestSuite(
+  () => {
+    test('findFirst', async () => {
+      const result = prisma.resource.findFirst({
+        where: { OrderBy: { gt: 0 } },
+        orderBy: { OrderBy: 'asc' },
+        cursor: { OrderBy: 1 },
+        distinct: ['OrderBy'],
+      })
+
+      await expect(result).resolves.toMatchInlineSnapshot(`null`)
+    })
+
+    test('findMany', async () => {
+      const result = prisma.resource.findMany({
+        where: { OrderBy: { gt: 0 } },
+        orderBy: { OrderBy: 'asc' },
+        cursor: { OrderBy: 1 },
+        distinct: ['OrderBy'],
+      })
```

Comment:
The chose nquery here is different than the original reproduction from the user: https://bitbucket.org/awimmjsystems/prisma/src/de00769d30bb07c78f797b2dfe6e1d4adaeef982/index.js#lines-8:11 Intentional or confirmed irrelevant?

Reply:
I confirmed that it was indeed failing first with this test, then made sure it does not fail on other levels and across other kinds of queries just to be more thorough.